packages/query-core/src/query.ts

```diff
@@ -392,6 +392,13 @@ export class Query<
         )
       }
       this.#abortSignalConsumed = false
+      if (this.options.persister) {
+        return this.options.persister(
+          this.options.queryFn,
+          queryFnContext as QueryFunctionContext<TQueryKey>,
+        )
+      }
+
       return this.options.queryFn(
         queryFnContext as QueryFunctionContext<TQueryKey>,
       )
```

Comment:
note: I think we would also need something for `infiniteQueries`, as they execute the QueryFn in a different branch:

https://github.com/TanStack/query/blob/8d5e08fdfffc42cadddd61865bc12f754b4b3564/packages/query-core/src/infiniteQueryBehavior.ts#L39-L45

Reply:
I think i have found a proper place to hook it up for infinite queries.
The only problem that is left is whenever you try to use new `createPersister` and `context` in `queryFn` types break for `queryFn`.